fe/fe-core/src/main/java/com/starrocks/sql/InsertPlannerV2.java

```diff
@@ -0,0 +1,892 @@
+import com.starrocks.sql.optimizer.transformer.RelationTransformer;
+import com.starrocks.sql.optimizer.transformer.SqlToScalarOperatorTranslator;
+import com.starrocks.sql.plan.ExecPlan;
+import com.starrocks.sql.plan.PlanFragmentBuilder;
+import com.starrocks.task.StreamLoadTask;
+import com.starrocks.thrift.TBrokerFileStatus;
+import com.starrocks.thrift.TPartitionType;
+import com.starrocks.thrift.TResultSinkType;
+import com.starrocks.thrift.TUniqueId;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class InsertPlannerV2 {
+    private static Logger LOG = LogManager.getLogger(InsertPlannerV2.class);
+    // Only for unit test
+    public static boolean enableSingleReplicationShuffle = false;
+    
+    private long loadJobId;
+    private TUniqueId loadId;
+    private long txnId;
+    private ConnectContext context;
+    private EtlJobType etlJobType;
+    private String timezone;
+    private boolean partialUpdate;
+    boolean isPrimaryKey;
+    boolean enableDictOptimize;
+    private long execMemLimit;
+    private long loadMemLimit;
+    private boolean strictMode;
+    private long timeoutS;
+    private int parallelInstanceNum;
+    private long startTime;
+    private long sqlMode;
+    private String loadTransmissionCompressionType;
+    private boolean enableReplicatedSorage;
+    private Map<String, String> sessionVariables;
+
+    private long dbId;
+    private Table destTable;
+    private DescriptorTable descTable;
+    private TupleDescriptor tupleDesc;
+    private List<Pair<Integer, ColumnDict>> globalDicts = Lists.newArrayList();
+
+    private List<PlanFragment> fragments = Lists.newArrayList();
+    private List<ScanNode> scanNodes = Lists.newArrayList();
+
+    // Broker Load related structs
+    private BrokerDesc brokerDesc;
+    private List<BrokerFileGroup> fileGroups;
+    private List<List<TBrokerFileStatus>> fileStatusesList;
+    private int filesAdded;
+    private Analyzer analyzer;
+
+    IdGenerator<PlanNodeId> planNodeGenerator = PlanNodeId.createGenerator();
+
+    // Insert into related structs
+    private InsertStmt insertStmt;
+
+    // Routine/Stream load related structs
+    List<ImportColumnDesc> columnDescs;
+    private StreamLoadTask streamLoadTask;
+    boolean routimeStreamLoadnegative;
+
+    public InsertPlannerV2(long loadJobId, TUniqueId loadId, long txnId, long dbId, OlapTable destTable,
+                            boolean strictMode, String timezone, long timeoutS,
+                            long startTime, boolean partialUpdate, ConnectContext context,
+                            Map<String, String> sessionVariables, long loadMemLimit, long execMemLimit, 
```

Comment:
It's better to unify to one mem limit

Reply:
To be consistent with the behaviour of broker load